### ISSUE TITLE
 Unable to prepare project.

### ISSUE THREAD

**Issue by dilanersej** [Microsoft/cordova-plugin-code-push#290]
Hello code-push team,

I've been a user of code push for almost a year without experincing any big issue. After couple of weeks of inactivity of our app we wanted to release some code updates.

When ever i run 

>  "code-push release-cordova "WMS-Android" android -d "Beta" 

before running the command i have executed

> cordova platform add android@5.2.1

I get the following error

> Unable to prepare project. Please ensure that the CWD represents a Cordova project and that the "android" platform was added by running "cordova platform add android".

These are the last 3 lines of the console whenever i run the above command 

> Prepared android project successfully
No scripts found for hook "after_prepare".
[Error]  Unable to prepare project. Please ensure that the CWD represents a Cordova project and that the "android" platform was added by running "cordova platform add android".

I've tried just running 

> cordova prepare android

Which works as supposed. I tried updateing nodejs,node, cordova to the latest versions without any luck,

Any idea what root of this problem is?

Kindly regards
Dilan

**Comment by max-mironov** [comment]
Hello @dilanersej, thanks for reaching us.
Could you please take a look at similar issues to see if it can help you somehow:
https://github.com/Microsoft/cordova-plugin-code-push/issues/133#issuecomment-223687871
https://github.com/Microsoft/cordova-plugin-code-push/issues/162#issuecomment-247489991
The fact that you are able to run `cordova prepare android` without any errors is looking a  bit strange, because as it was metioned in this threads code-push CLI simply calls this command under the hood. 
Is it possible for you to verify if you could see any issues with fresh new cordova project, so we can determine if the issue is with your app only or there is something wrong with your environment?

**Comment by dilanersej** [comment]
Hi @max-mironov I tried looking at the links you have provided unfortunatly it does not seem to be same issue as I have.

I tried creating a clean cordova project and tried to push code changes to code-push server with succes.

So something must have changed in my project since my last push. Do you have any idea where to look? Mean while I will investigate this problem further and report back.

Kindly regards
Dilan

**Comment by max-mironov** [comment]
@dilanersej 
Well, so `cordova prepare android` runs without any issues, right? 
It is really strange, please take a look at the [following snippet](https://github.com/Microsoft/code-push/blob/master/cli/script/command-executor.ts#L1267):
```
try
 {
                execSync([cordovaCLI, cordovaCommand, platform, "--verbose"].join(" "), { stdio: "inherit" });
            } catch (error) {
                throw new Error(`Unable to ${cordovaCommand} project. Please ensure that the CWD represents a Cordova project and that the "${platform}" platform was added by running "${cordovaCLI} platform add ${platform}".`);
            }
```
It doesn't do anymore rather then executing `cordova build android --verbose`.
So if you are able to run this command from cmd you should never see this issue in theory :)
Probably you could share with me (v-maximi at microsoft.com) your app and I can build it on my machine to see if the issue persist?

**Comment by dilanersej** [comment]
@max-mironov 
Problem solved! So i noticed one of my plugins could not be retrieved from github which resulted in a warning, I did not think further about this since its a plugin that I do not use anymore (Ye i know i should clean up). Anyways I decided to remove the plugin to build and push the changes to code-push server, this time it was a succes!

Thanks for your help!

Kindly regards,
Dilan

**Comment by max-mironov** [comment]
@dilanersej  I'm happy to hear that things are working finally and thanks so much for letting us know!

**Comment by talavidan4** [comment]
I ran into the same issue today, when trying to deploy both for ios and android.

After i made my last deployment almost 2 month from now,
I failed to run the one line command that recommended by the docs:
`appcenter codepush release-cordova -a <ownerName>/MyApp-iOS ios`.

The error was as mention above:
> Error: Unable to prepare project. Please ensure that the CWD represents a Cordova project and that t
he "android" platform was added by running "cordova platform add android".

I manage to deploy to code push appcenter server, by doing this with the two command line version:
`cordova prepare ios
appcenter codepush release -a <ownerName>/MyApp-iOS -c ./platforms/ios/www -t 1.0.0`

No idea why, till now it worked perfect..
